### ISSUE TITLE
Improve troubleshooting docs

### ISSUE THREAD

**Issue by AlexDBlack** [eclipse/deeplearning4j#1390]
Here's some improvements / things to answer / questions that have come up more than once:
- How to do train/test split?
- How to deal with normalization at train vs. test time
- How many layers?
- What architecture?
- How to check if the normalization is correct?
- Improve loss function section: what configuration option specifically?
- Are there certain problems that _can't_ be solved using neural networks?
- (When) should I use an ensemble?
- Do I need to provide a .weightInit(...) on _every_ layer, or just once at the start?
- Issue with minibatch not containing proper shuffling of examples (and: how to check?)

Also, add links to:
- RBM tuning guides
- Links to resources such as cs231n and deeplearningbook.org

**Comment by tomthetrainer** [comment]
Here's some improvements / things to answer / questions that have come up more than once:

* How to do train/test split?
** note add to ETL guide

* How to deal with normalization at train vs. test time
** this is covered in an example, link to that add to etl guide

* How many layers?
** Enough, I do not know how to answer that. Maybe in choosing a network page I can address that **

* What architecture?
** see if we have choosing a neural net page, I think it is out of date, update

* How to check if the normalization is correct?
** I do a brute force check and look at actual values, I suppose sum() should return 0, and some other numerical checks could be run. Or perhaps by correct they mean the same. SaveNormalizer needs to be documented. 

* Improve loss function section: what configuration option specifically?
** These sections have been hit, review to verify


* Are there certain problems that can't be solved using neural networks?
** Theoretically no, they make easy problems hard and hard problems easy. 

* (When) should I use an ensemble?
** 

* Do I need to provide a .weightInit(...) on every layer, or just once at the start?

* Issue with minibatch not containing proper shuffling of examples (and: how to check?)
** too narrow I may ignore

Also, add links to:

* RBM tuning guides
* Links to resources such as cs231n and deeplearningbook.org

**Comment by huitseeker** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [DataVec, workflow & al. - improve documentation](https://trello.com/c/3u3jt2lF/5-datavec-workflow-al-improve-documentation)

**Comment by agibsonccc** [comment]
Closing stale issue: cleanup.
